### PR TITLE
New design docs

### PR DESCRIPTION
**Pull request by Lydiaofficial**
Proposal for new documentation for contributing as a new designer.

The proposed changes include new sidebar structure and new text documents.

### CODE REVIEW

**Comment by wetneb** [comment]
Thanks @Lydiaofficial! It looks like the website cannot be built with those changes because the sidebar.js contains incorrect page identifiers. See [the logs](https://app.netlify.com/sites/openrefine-website/deploys/64f8600b3c80e10008fc920f):

```
1:19:07 PM: [ERROR] Error: Invalid sidebar file at sidebars.js.
1:19:07 PM: These sidebar document ids do not exist:
1:19:07 PM: - technical-reference/creating-github-issues
1:19:07 PM: - technical-reference/design-contributions-workflows
1:19:07 PM: - technical-reference/openrefine-design-system
1:19:07 PM: - technical-reference/providing-design-assets-and-design-implementation
1:19:07 PM: 
1:19:07 PM: Available document ids are:
1:19:07 PM: - index
1:19:07 PM: - manual/cellediting
1:19:07 PM: - manual/columnediting
1:19:07 PM: - manual/exploring
1:19:07 PM: - manual/exporting
1:19:07 PM: - manual/expressions
1:19:07 PM: - manual/facets
1:19:07 PM: - manual/grel
1:19:07 PM: - manual/grelfunctions
1:19:07 PM: - manual/installing
1:19:07 PM: - manual/jythonclojure
1:19:07 PM: - manual/reconciling
1:19:07 PM: - manual/running
1:19:07 PM: - manual/sortview
1:19:07 PM: - manual/starting
1:19:07 PM: - manual/transforming
1:19:07 PM: - manual/transposing
1:19:07 PM: - manual/troubleshooting
1:19:07 PM: - manual/wikibase/advanced-schemas
1:19:07 PM: - manual/wikibase/configuration
1:19:07 PM: - manual/wikibase/new-entities
1:19:07 PM: - manual/wikibase/overview
1:19:07 PM: - manual/wikibase/quality-assurance
1:19:07 PM: - manual/wikibase/reconciling
1:19:07 PM: - manual/wikibase/schema-alignment
1:19:07 PM: - manual/wikibase/uploading
1:19:07 PM: - technical-reference/architecture-4
1:19:07 PM: - technical-reference/architecture-before-4
1:19:07 PM: - technical-reference/brainstorming-and-ideation
1:19:07 PM: - technical-reference/build-test-run
1:19:07 PM: - technical-reference/clustering-in-depth
1:19:07 PM: - technical-reference/code-contributions
1:19:07 PM: - technical-reference/contributing
1:19:07 PM: - technical-reference/creating-an-issue
1:19:07 PM: - technical-reference/design-contribution-workflows
1:19:07 PM: - technical-reference/design-contributions
1:19:07 PM: - technical-reference/development-roadmap
1:19:07 PM: - technical-reference/documentation-contributions
1:19:07 PM: - technical-reference/feedback-and-prioritization
1:19:07 PM: - technical-reference/functional-tests
1:19:07 PM: - technical-reference/get-involved
1:19:07 PM: - technical-reference/homebrew-cask-process
1:19:07 PM: - technical-reference/introduction
1:19:07 PM: - technical-reference/maintainer-guidelines
1:19:07 PM: - technical-reference/migrating-older-extensions
1:19:07 PM: - technical-reference/openRefine-design-system
1:19:07 PM: - technical-reference/openrefine-api
1:19:07 PM: - technical-reference/providing-design-assets-and-implementation
1:19:07 PM: - technical-reference/reconciliation-api
1:19:07 PM: - technical-reference/resources
1:19:07 PM: - technical-reference/translating-ui
1:19:07 PM: - technical-reference/version-release-process
1:19:07 PM: - technical-reference/why-get-involved
1:19:07 PM: - technical-reference/wikibase/architecture
1:19:07 PM: - technical-reference/writing-extensions
```

You should be able to fix this by updating the `sidebar.js` on your side with the correct page identifiers, and committing / pushing to update this pull request. Once that is done, we will have a preview of the website with your changes, which should make it easier to review.
Let me know if you need help for any of those steps.

**Comment by Lydiaofficial** [comment]
@wetneb I updated the identifiers but they failed again. Feels like I'm doing a whole lot wrong and I can't see the logs.

**Comment by wetneb** [comment]
To view the logs, go to the "All checks have failed" section of this page, and click on the "Details" link to the right-hand side of a failing check. This will take you to the logs.

It looks like you have successfully solved the first errors. Now the current logs say:
```
Error: Image docs/static/img/openrefine-design-project.png used in docs/technical-reference/design-contribution-workflows.md not found.
```

**Comment by Lydiaofficial** [comment]
This is what I see when I click to view logs 
Screenshot 2023-09-07 060648

Also, could you check if the image error has been fixed now?

**Comment by wetneb** [comment]
It is curious that you cannot view the logs… Maybe it's something that's region-specific, depending on some content delivery network?

Anyway, your commit which was meant to update the image path only added a newline:
https://github.com/OpenRefine/openrefine.org/pull/240/commits/1214bdf7b7a30e4a787880c0d9c37efbd51d723a

So the deploy logs still say:
```
3:03:02 PM: Error: Image docs/static/img/openrefine-design-project.png used in docs/technical-reference/design-contribution-workflows.md not found.
```

You could also generate the site locally, on your machine: that would make it easier for you to see the errors without having to rely on those cloud services. We have documentation about how to set that up, but I just realized that it's really not that well written and quite out of date, so I have made a new version of it:
https://deploy-preview-243--openrefine-website.netlify.app/docs/technical-reference/documentation-contributions
If you have feedback about this documentation, feel very invited to voice it in the corresponding pull request: #243.

**Comment by magdmartin** [comment]
@Lydiaofficial, thanks for the PR and putting all this together. It will help tremendously when welcoming new contributors (including non-designers). 
Since there is a lot of content here, what would be the best way to provide feedback? I can do a first review of the structure of the guide first. Would that work for you?

File: docs/technical-reference/design-contribution-workflows.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+title: Design Contribution Workflows
+sidebar_label: Design Contribution Workflows
+--- 
+
+### Identifying design issues
+
+The OpenRefine project contains several repositories, but you’re most likely to find design-related issues in the main application repository here: https://github.com/OpenRefine/OpenRefine. Alternatively, further design issues might be linked to the main [OpenRefine website repository](https://github.com/OpenRefine/openrefine.org), or some additional extensions. 
+
+Within OpenRefine's GitHub, the [OpenRefine Design](https://github.com/orgs/OpenRefine/projects/1) Project is the main entry point for designers to identify and select suitable issues to work on.
+![This is a screenshot showing the design project in the OpenRefine GitHub repository](<../static/img/openrefine-design-project.png>)
```
Comment (wetneb) [inline docs/technical-reference/design-contribution-workflows.md:12]: It's not really clear to me that this screenshot is useful - isn't it enough to give a link to the project (as you already did)?

Concerning the syntax to embed a URL, the simplest way to figure out the correct syntax is to look at examples from other pages in the documentation. According to those examples, you could try this:
```suggestion
![This is a screenshot showing the design project in the OpenRefine GitHub repository](/img/openrefine-design-project.png)
```

File: docs/technical-reference/design-contribution-workflows.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+
+The OpenRefine design team uses  GitHub Projects to effectively manage and track design-related issues across all relevant repositories. GitHub Projects allows us to group issues that have been tagged with a design-related theme and track progress from start to finish.
+
+![This screenshot shows the OpenRefine design project in list view](<../static/img/project-list-view.png>)
```
Comment (wetneb) [inline docs/technical-reference/design-contribution-workflows.md:17]: Similarly, you could try out this format:
```suggestion
![This screenshot shows the OpenRefine design project in list view](/img/project-list-view.png)
```

**Comment by Lydiaofficial** [comment]
> @Lydiaofficial, thanks for the PR and putting all this together. It will help tremendously when welcoming new contributors (including non-designers). Since there is a lot of content here, what would be the best way to provide feedback? I can do a first review of the structure of the guide first. Would that work for you?

Yeah sure. That works just fine.

File: docs/technical-reference/design-contribution-workflows.md (hunk 1 above)
Comment (Lydiaofficial) [reply docs/technical-reference/design-contribution-workflows.md:12]: That is brilliant. Thank you!

**Comment by lozanaross** [comment]
> @Lydiaofficial, thanks for the PR and putting all this together. It will help tremendously when welcoming new contributors (including non-designers). Since there is a lot of content here, what would be the best way to provide feedback? I can do a first review of the structure of the guide first. Would that work for you?

@magdmartin - just also FYI, the doc that these proposed changes are based on was checked and edited over multiple rounds by myself and Jan Dietrich, so we hope the structure is already quite lean & checked by several people. Of course suggestions for improvement are always welcome. You can also comment on the [google doc](https://docs.google.com/document/d/1H39Nr6CeXmAvHnh8jqobXXP_d4tCqrzkhGPR9RBlNp4/edit?usp=sharing), if that's faster / easier (vs the multiple, split md files).

As to further changes beyond the scope of 'contributing as a designer', I am aware there are plans for these, but I hope the changes we propose with the sidebar structure here are a good short-term solution until all other changes to the contribution's docs are prepped (eg contributing as a trainer, translator, etc).

**Comment by wetneb** [comment]
> @magdmartin - just also FYI, the doc that these proposed changes are based on was checked and edited over multiple rounds by myself and Jan Dietrich, so we hope the structure is already quite lean & checked by several people.

No matter who earmarked the current proposal I think everyone should feel welcome to make comments and suggest changes, even fundamental ones. That's what PRs are for :)

**Comment by lozanaross** [comment]
> > @magdmartin - just also FYI, the doc that these proposed changes are based on was checked and edited over multiple rounds by myself and Jan Dietrich, so we hope the structure is already quite lean & checked by several people.
> 
> No matter who stamped on the current proposal I think everyone should feel welcome to make comments and suggest changes, even fundamental ones. That's what PRs are for :)

Of course, I just wanted to make it clear that the important part for the Outreachy deliverables from Lydia are the designer contribution docs, those are the things that were checked at least once. The rest of the changes in the PR - i.e. the sidebar - is not a perfect solution, but just a way to publish the Outreachy work in some form. I am not sure if it was correct to bundle everything in one go, but this was the first time trying to do a design contribution to the site, so space for lessons to be learned :-)

**Comment by magdmartin** [comment]
@lozanaross, I appreciate the background information you provided on how the document was prepared. 

@Lydiaofficial,
In my review below, I focused on the overall structure of the documentation and suggested ways to align it better with existing guidelines.

In addition to my comments, I would like to suggest that we revisit the following two pages as part of this PR:
- https://openrefine.org/docs/technical-reference/contributing (link to the design section)
- https://openrefine.org/docs/technical-reference/design-contributions

Please let me know if you have any questions or concerns.

File: docs/technical-reference/Why-Get-Involved.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+---
+id: why-get-involved
+title: Why Get Involved
+sidebar_label: Why Get Involved
+---
+
+OpenRefine is a powerful open-source tool for cleaning, transforming, and enriching data. Originally developed by Google, it was released open source in 20xx and is now maintained by a diverse international community. Users like journalists, librarians and researchers value it for its powerful capabilities to work with messy data.
```
[the remark this answers is not in the document]
Comment (lozanaross) [reply docs/technical-reference/Why-Get-Involved.md:7]: @Lydiaofficial can you make the update from "20xx" to "2013"

File: docs/technical-reference/Feedback-and-prioritization.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+---
+id: feedback-and-prioritization
+title: Feedback and Prioritization
+sidebar_label: Feedback and Prioritization
+---
+
+When engaging in design discussions and reviews, share your work within GitHub issues to gather feedback on usability, aesthetics, and project alignment. Incorporate received feedback by refining your designs, ensuring they adhere to the user goals identified in the original formulation of the issue. 
+
+Collaborate with maintainers and contributors to prioritize the order of addressing and implementing design tasks, considering the community's size and potential competing responsibilities. Be aware that feedback from project maintainers and experienced contributors might take time due to the community's small scale and other ongoing tasks.
```
[the remark this answers is not in the document]
Comment (lozanaross) [reply docs/technical-reference/Feedback-and-prioritization.md:9]: Sure, @Lydiaofficial - can you add the following at the end of this section: "For the same reason, implementation might also take time. The asynchronous feedback and implementation process does not remove any value from the design suggestion as it helps elicit future functionalities or improvements."

File: docs/technical-reference/Why-Get-Involved.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
+---
+id: why-get-involved
+title: Why Get Involved
```
[the remark this answers is not in the document]
Comment (lozanaross) [reply docs/technical-reference/Why-Get-Involved.md:3]: Sure, that sounds good to me. @Lydiaofficial can you make the change in the sidebars.js structure?

**Review by magdmartin — APPROVED** [review]
Thank you @Lydiaofficial, for the update and @lozanaross for the feedback. I am confident this document will be useful for current and future contributors, even beyond designers. 

I am good to merge this PR as-is so we can all benefit from a publicly documented process. We can make subsequent copy editing adjustments in separate PR.

**Comment by Lydiaofficial** [comment]
Thank you, @magdmartin

**Comment by magdmartin** [comment]
@lozanaross and @wetneb, just wanted to give a friendly reminder to please take a look at this PR when you have a moment. Merging it quickly would really help out with our ongoing designer recruitment efforts. Thank you!

**Pull request merged**